pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -421,6 +421,11 @@ func (r *shootReconciler) runReconcileShootFlow(ctx context.Context, o *operatio
 			Fn:           flow.TaskFn(botanist.DeployIngressDNSResources).DoIf(!o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(nginxLBReady),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Cleaning up orphaned DNSRecord secrets",
+			Fn:           flow.TaskFn(botanist.CleanupOrphanedDNSRecordSecrets),
```

Comment:
I think this should only be executed if the cluster is not hibernated, since deploying some of the DNS records is also constrained by the same condition.
```suggestion
			Fn:           flow.TaskFn(botanist.CleanupOrphanedDNSRecordSecrets).DoIf(!o.Shoot.HibernationEnabled),
```

Reply:
done.